src/spatialmath/utils.rs

```diff
@@ -48,22 +51,306 @@ impl EulerAngles {
 #[repr(C)]
 #[derive(Clone, Copy, Debug)]
 pub struct AxisAngle {
-    pub vector: Vector3<f64>,
+    pub axis: Vector3<f64>,
     pub theta: f64
 }
 
 impl AxisAngle {
     pub fn new(x: f64, y: f64, z: f64, theta: f64) -> Self {
-        AxisAngle { vector: Vector3::new(x, y, z), theta }
+        AxisAngle { axis: Vector3::new(x, y, z), theta }
+    }
+}
+
+fn orientation_vector_theta_from_rotated_axes(
+    new_x: Quaternion<f64>, new_z: Quaternion<f64>
+) -> f64 {
+    if 1.0 - new_z.k.abs() > ANGLE_ACCEPTANCE {
+        let new_z_imag = new_z.imag();
+        let new_x_imag = new_x.imag();
+        let z_imag_axis = Vector3::z_axis();
+
+        let normal_1 = new_z_imag.cross(&new_x_imag);
+        let normal_2 = new_z_imag.cross(&z_imag_axis);
+        let cos_theta_cand = normal_1.dot(&normal_2) / (normal_1.norm() * normal_2.norm());
+        let cos_theta = match cos_theta_cand {
+            val if val < -1.0 => -1.0,
+            val if val > 1.0 => 1.0,
+            _ => cos_theta_cand
+        };
+
+        return match cos_theta.acos() {
+            val if val > ANGLE_ACCEPTANCE => {
+                let new_z_imag_unit = UnitVector3::new_normalize(new_z_imag);
+                let rot_quat_unit = UnitQuaternion::from_axis_angle(&new_z_imag_unit, -1.0 * val);
+                let rot_quat = rot_quat_unit.quaternion();
+                let z_axis_quat = Quaternion::new(0.0, 0.0, 0.0, 1.0);
+                let test_z = (rot_quat * z_axis_quat) * rot_quat.conjugate();
+                let test_z_imag = test_z.imag();
+
+                let normal_3 = new_z_imag.cross(&test_z_imag);
+                let cos_test = normal_1.dot(&normal_3) / (normal_3.norm() * normal_1.norm());
+                match cos_test {
+                    val2 if (1.0 - val2) < (ANGLE_ACCEPTANCE * ANGLE_ACCEPTANCE) => -1.0 * val,
+                    _ => val
+                }
+            },
+            _ => 0.0
+        };
+    }
+    match new_z.k {
+        val if val < 0.0 => -1.0 * new_x.j.atan2(new_x.i),
+        _ => -1.0 * new_x.j.atan2(new_x.i * -1.0)
+    }
+}
+
+#[repr(C)]
+#[derive(Clone, Copy, Debug)]
+pub struct OrientationVector {
+    pub o_vector: UnitVector3<f64>,
+    pub theta: f64
+}
+
+impl OrientationVector {
+    pub fn new(o_x: f64, o_y: f64, o_z: f64, theta: f64) -> Self {
+        let o_vector = UnitVector3::new_normalize(Vector3::new(o_x, o_y, o_z));
+        OrientationVector{o_vector, theta}
+    }
+
+    pub fn from_quaternion(quat: &Quaternion<f64>) -> Self {
```

Comment:
could you also implement the To and From traits?

Reply:
I can do it for the custom structs I created, but for quaternion and rotation matrix I would have to create wrappers. Is it fine to skip those two?